packages/frontend/src/utils/getVolunteers.ts

```diff
@@ -0,0 +1,52 @@
+import { FirebaseCollection } from 'uva-shared';
+import { firebaseDb } from 'src/main';
+import { IVolunteer } from 'uva-shared/src/types/volunteer';
+import { collection, getDocs, limit, orderBy, query, startAfter } from 'firebase/firestore';
+import { QueryArgs } from './getEvents';
+
+export interface GetVolunteersResult {
+  readonly volunteers: IVolunteer[];
+  readonly lastVolunteer: DocumentSnapshot | null;
+}
+
+import { DocumentSnapshot } from 'firebase/firestore';
+
+export const getVolunteers = async (
+  name?: string,
+  lastVolunteer?: DocumentSnapshot | null,
+  pageSize = 10,
+): Promise<GetVolunteersResult> => {
+  const queryArgs: QueryArgs[] = [orderBy('firstName', 'desc'), limit(pageSize)];
+
+  if (lastVolunteer) {
+    queryArgs.push(startAfter(lastVolunteer));
+  }
+
+  const queryForVolunteers = query(collection(firebaseDb, FirebaseCollection.Volunteers), ...queryArgs);
+
+  try {
+    const querySnapshot = await getDocs(queryForVolunteers);
```

Comment:
This won't work well, filter by name should also be part of query to firebase. 

For example, if there are [{firstName: 'a'}, {firstName: 'b'}, {firstName: 'c'}, {firstName: 'd'}] and pageSize = 2, if you'll try to search name 'a', you'll get empty result, because firebase will return two sorted desc names:  [{firstName: 'c'}, {firstName: 'd'}] and it will be filtered by condition firstName === 'a'

You can use approach from https://stackoverflow.com/questions/46573804/firestore-query-documents-startswith-a-string to form query that starts with name.toLowerCase(). Also to make the search case insensitive, we need to add firstNameLowerCase and lastNameLowerCase fields to Volonteer

Reply:
Refactored and commited